pkg/gateway/errors/errors.go

```diff
@@ -176,20 +176,24 @@ const (
 
 type errorCodeMap map[APIErrorCode]APIError
 
-func (e errorCodeMap) ToAPIErr(errCode APIErrorCode) APIError {
+func (e errorCodeMap) ToAPIErr(errCode APIErrorCode, err error) APIError {
```

Comment:
Maybe add a function 
```
func (e errorCodeMap) ToAPIErrWithInternalError(errCode APIErrorCode, err error) APIError {
``` 
It could leverage the existing `ToAPIErr` func.
That way you won't have dozens of calls to `ToAPIErr` with `nil` as the err, although an actual error does exist. I also wonder what the next caller to this funciton would do, pass an `err` or `nil`?

Reply:
Done